### PR TITLE
Refactor inlined JavaScript/CSS styles to external files #38

### PR DESCRIPTION
**Pull request by mp911de**
Move scripts and styles to own files. Use module exports for JavaScript and organize files according to the referencing page. Refactor main-layout JavaScript/CSS assets. Simplify footer definition. Adopt security config for CSS/JS resources.

### CODE REVIEW

**Comment by rwinch** [comment]
Thanks for the PR!

I'm wondering if we can move all the resources into a common folder so as to map only a single folder. Perhaps we could restructure to use static/assets/css and static/assets/js folders. Then we can map only /assets/*\* to public.

Part of the concern is mapping file extensions to security constraints is fragile due to the way Spring MVC handles file extensions. For example, I have a `@RequestMapping("/admin/cla/link")` for my controller. However, the URL http://localhost:8080/admin/cla/link.css will resolve to the controller (notice that this file extension would match `../**/*.css` thus giving un authorized access to the user.

File: src/main/resources/templates/layouts/main.html [hunk 1]
```diff
@@ -123,15 +96,9 @@
...
-				$('#sign-out').click(function(e) {
-						e.preventDefault();
-						$('#sign-out-form').submit();
-				});
-		});
-		//-->
-		</script>
+    	<script th:src="@{/layouts/main.js}" type="text/javascript"></script>
+    	<script type="text/javascript">
+			mainLayout.initialize();
```
Comment (rwinch) [inline src/main/resources/templates/layouts/main.html]: Any chance we can externalize this too? I'd like to externalize everything so we can setup a Content Security Policy http://www.html5rocks.com/en/tutorials/security/content-security-policy/

File: src/main/resources/templates/cla/ccla/sign.html [hunk 2]
```diff
@@ -126,10 +126,12 @@
 				<button id="sign" type="submit" class="btn btn-success">Submit & Contribute</button>
 			</form>
 		</div>
+
 		<div layout:fragment="javascript">
+            <script th:src="@{/pages/cla/ccla/sign.js}" type="text/javascript"></script>
 			<script type="text/javascript">
-			$("[data-toggle=popover]").popover({html:true});
+				claCclaSign.initialize();
```
Comment (rwinch) [inline src/main/resources/templates/cla/ccla/sign.html]: Any chance we can externalize this too? I'd like to externalize everything so we can setup a Content Security Policy http://www.html5rocks.com/en/tutorials/security/content-security-policy/

File: src/main/resources/templates/admin/cla/link.html [hunk 3]
```diff
@@ -147,44 +135,13 @@ <h4 class="modal-title">What does this do?</h4>
...
-								$('#reload-repo-list').click(reloadRepositoryList);
-								$('#open-hook-urls').click(openDataUrls);
-								$('#contributing-md-urls').click(openDataUrls);
-								$('#contributing-adoc-urls').click(openDataUrls);
-
-						</script>
-
+			<script th:src="@{/pages/admin/cla/link.js}" type="text/javascript"></script>
+			<script type="text/javascript">
+				adminClaLink.initialize();
```
Comment (rwinch) [inline src/main/resources/templates/admin/cla/link.html]: Any chance we can externalize this too? I'd like to externalize everything so we can setup a Content Security Policy http://www.html5rocks.com/en/tutorials/security/content-security-policy/

File: src/main/resources/templates/admin/cla/link.html [hunk 4]
```diff
@@ -3,21 +3,9 @@
...
-		height: 33px;
-}
-
-.select2-container--default .select2-selection--multiple .select2-selection__choice {
-		background-color: #03786e;
-		border: 1px solid #03786e;
-		color: #fff;
-}
-		</style>
+        <link th:href="@{/webjars/select2/dist/css/select2.min.css}" rel="stylesheet"/>
```
Comment (rwinch) [inline src/main/resources/templates/admin/cla/link.html]: Not a huge deal (I can clean this up), but tabs not spaces please

File: src/main/resources/templates/layouts/main.html (hunk 1 above)
Comment (mp911de) [inline src/main/resources/templates/layouts/main.html]: Sure, I'll adopt that for the other inline code snippets.

**Comment by mp911de** [comment]
Thanks for your feedback. I'll take a look and update the PR.

**Comment by mp911de** [comment]
PR updated

**Comment by rwinch** [comment]
Thanks for the PR! This is now merged into master

**Pull request merged**